cargo-shuttle/src/lib.rs

```diff
@@ -339,3 +358,41 @@ impl Shuttle {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::Shuttle;
+    use std::path::PathBuf;
+    use crate::args::ProjectArgs;
+
+    fn path_from_workspace_root(path: &str) -> PathBuf {
+        PathBuf::from(std::env::var("CARGO_MANIFEST_DIR").unwrap())
+            .join("..")
+            .join(path)
+    }
+
+    #[test]
+    fn find_root_directory_returns_proper_directory() {
+        let working_directory = path_from_workspace_root("examples/axum/hello-world/src");
+        let deep_working_directory = path_from_workspace_root("examples/axum/hello-world/target/debug");
+
+        let root_dir = Shuttle::find_root_directory(&working_directory).unwrap();
+        let deep_root_dir = Shuttle::find_root_directory(&deep_working_directory).unwrap();
+
+        assert_eq!(root_dir, path_from_workspace_root("examples/axum/hello-world/"));
+        assert_eq!(deep_root_dir, path_from_workspace_root("examples/axum/hello-world/"));
+    }
```

Comment:
Rather make this two tests.

Reply:
Sure, will separate them!

re: failing if the `target` does not exist. I assumed that no one would remove the `examples` folder. If someone does that, then the other tests will fail as well since we run the tests against `examples/axum`. And iirc, the examples always come with the `target` folder already inside, no?

Reply:
By the way, the test/variable naming is horrible in that test. Please advise something better, my brain malfunctioned.

```rust

#[test]
    fn find_root_directory_returns_proper_directory() {
        let working_directory = path_from_workspace_root("examples/axum/hello-world/src");

        let root_dir = Shuttle::find_root_directory(&working_directory).unwrap();

        assert_eq!(
            root_dir,
            path_from_workspace_root("examples/axum/hello-world/")
        );
    }

    #[test]
    fn find_root_directory_returns_proper_directory_from_a_deeper_directory() {
        let deep_working_directory =
            path_from_workspace_root("examples/axum/hello-world/target/debug");

        let deep_root_dir = Shuttle::find_root_directory(&deep_working_directory).unwrap();

        assert_eq!(
            deep_root_dir,
            path_from_workspace_root("examples/axum/hello-world/")
        );
    }

```

Reply:
> By the way, the test/variable naming is horrible in that test. Please advise something better, my brain malfunctioned.

I don't mind the variable names. They can even be the same in both tests...

Reply:
Excerpt from the Discord conversation:
```
@ivancernja : re: target folder not being there for the tests. I can run it against rocket/authentication/src which is deeper than hello_world/src -- do you think that makes sense or should we scratch that test?

@chesedo : Hmm, we can scratch it if there isn't a deeply nested folder in the examples
```

Removing the `deeper` test for now.